lib/tree_sitter/node.rb

```diff
@@ -166,21 +166,14 @@ def to_a
     #
     # See {#fetch_all}.
     def fetch(*keys, all: false, **_kwargs)
-      dict = {}
-      keys.each.with_index do |k, i|
-        dict[k.to_s] = i
-      end
+      key_set = keys.to_set(&:to_s)
+      fields = {}
+      each_field do |f, _c|
+        fields[f] = self[f] if key_set.delete(f)
 
-      res = {}
-      each_field do |f, c|
-        if dict.key?(f)
-          res[f] = c
-          dict.delete(f)
-        end
-        break if dict.empty?
+        break if key_set.empty?
       end
-
-      res = keys.uniq.map { |k| res[k.to_s] }
+      res = keys.uniq.map { |k| fields[k.to_s] }
```

Comment:
I would not leave the uniq, it creates useless surprise IMHO.  And btw I would just

```ruby
res = fields.values_at(*keys)
```

Reply:
That was the bug I fixed and still not pushed :)